### PR TITLE
added unique collectors column in spotlight

### PR DESCRIPTION
**Pull request by lohba**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇  _ Do a quick check before the merge. 

### PR type
- [ ] Bugfix
- [X] Feature
- [ ] Refactoring

### Before submitting Pull Request, please make sure:
- [X] My contribution builds **clean without any errors or warnings**
- [X] I've merged recent default branch -- **main** and I've no conflicts
- [X] I've tried respect high code quality standards
- [X] I've didn't break any original functionality
- [X] I've posted screenshot of demonstrated change in this PR

### Optional
- [ ] I've tested it at </rmrk/collection/26902bc2f7c20c546a-1FVG7>
- [ ] I've tested PR on mobile and everything works
- [ ] I found edge cases

### What's new?
- [X] PR closes ##1455
- [X] Added Unique Collectors Column 

### Had issue bounty label ?
- [X] Fill up your KSM address: [Payout](https://kodadot.xyz/transfer/?target=EAeSkdwjgMBmVFcACVFGzqxmGqib8k8uXdNmtWTcWsLoGPU)

### Community participation
- [X] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot
- [X] My fix has changed **something** on UI, a screenshot for others, is best to understand changes.
Screen Shot 2021-12-12 at 9 32 22 AM

### CODE REVIEW

**Comment by yangwao** [comment]
Curious, is there is mathematical annotation for unique? @vikiival @roiLeo  
If so, @lohba let's add it and add an explanation tooltip (don't forget put translations to en.json) and I can raise bounty! :)

**Review by roiLeo — CHANGES_REQUESTED** [review]
We just need to sync with main branch as it has already been done.
I'm not a fan of `uniqueCollectorFn` because it will remove nft that belongs to issuer.

> Curious, is there is mathematical annotation for unique? @vikiival @roiLeo If so, @lohba let's add it and add an explanation tooltip (don't forget put translations to en.json) and I can raise bounty! :)

From Wikipedia, it's "∃!"

**Comment by lohba** [comment]
Thanks for the feedback @roiLeo @yangwao. I ended up using `uniqueCollectorFn` because while trying to reuse the unique collection number from CollectionItem, which queries from `nftListByCollection`,  SpotflightTable queries from `collectionIssuerList`. Is there a simple way to sync this up?

**Comment by roiLeo** [comment]
> Thanks for the feedback @roiLeo @yangwao. I ended up using `uniqueCollectorFn` because while trying to reuse the unique collection number from CollectionItem, which queries from `nftListByCollection`, SpotflightTable queries from `collectionIssuerList`. Is there a simple way to sync this up?

You could try to open main branch on another window and compare the files in `components/spotlight`
Currently, I see that we miss Volume data on main-nuxt

**Comment by lohba** [comment]
@roiLeo Sorry, I'm a bit confused. I'm not seeing unique collectors on the main branch. The only difference I see related to volume which I don't think is related.

**Comment by roiLeo** [comment]
@lohba I got confused with `uniqueCollectors`  function in "series-insight" which is not the same.

Could you try to add Volume Column too?

**Comment by lohba** [comment]
Added Volume column in spotlight 

Screen Shot 2021-12-15 at 11 58 35 AM

**Comment by yangwao** [comment]
Is this good to go? @lohba @roiLeo ?

**Comment by yangwao** [comment]
@lohba let's change unique to ∃! then and unique collectors to ∃! collectors plus add there tooltip like it's on the score column with explanation unique and unique collectors :)

**Comment by lohba** [comment]
Ok, will do - what is the definition of "Unique"?

**Comment by yangwao** [comment]
> @yangwao Ok, will do - what is the difference between unique and unique collectors?

Oh good point, it's unique NFTs of an artist so thinking what should be better terminology here, maybe unique items would be best if no one will come something better 
It's counted by metadata which is might be bit skewed when I'm thinking as simple mint introduced other way around hmm

File: langDir/en.json [hunk 1]
```diff
@@ -439,7 +439,8 @@
     "count": "Σ collection",
     "collectors": "Collectors",
     "score": "Score",
-    "empty": "Empty"
+    "empty": "Empty",
+    "uniqueCollectors": "Unique Collectors"
```
Comment (roiLeo) [inline langDir/en.json:443]:
```suggestion
    "uniqueCollectors": "∃! collectors"
```

File: components/spotlight/utils.ts [hunk 2]
```diff
@@ -19,6 +20,10 @@ export const nftFn = (a: any): Row => {
   const sold = a.nfts.nodes.reduce(soldFn, 0)
   const unique = a.nfts.nodes.reduce(uniqueFn, new Set()).size
   const total = a.nfts.totalCount
+  const uniqueCollectors = a.nfts.nodes.reduce(uniqueCollectorFn, new Set()).size
+  const buyEvents = a.nfts.nodes.map(onlyEvents).map(pairListBuyEvent).flat()
+  const volume =  Number(getVolume(buyEvents))
+  console.log("volume " + volume)
```
Comment (roiLeo) [inline components/spotlight/utils.ts:26]:
```suggestion
```

**Review by roiLeo — CHANGES_REQUESTED** [review]

**Comment by lohba** [comment]
unique items and unique collectors tooltip label 

Screen Shot 2021-12-20 at 4 50 41 PM

**Review by roiLeo — APPROVED** [review]
lgtm, remove trailing space before merge

**Comment by lohba** [comment]
@yangwao

**Comment by yangwao** [comment]
I've sent a bit more as you put more effort here and we reward extra effort ;)

Perfect, I’ve sent the payout https://kusama.subscan.io/extrinsic/0xcf5c51a07c8fe686e6a6ffbb8da8b5489b7a3abb30e93e4183d797a5e7a029ea

Let’s grab another issue and get rewarded! 
https://github.com/kodadot/nft-gallery/issues/

**Pull request merged**